cli/src/cli.rs

```diff
@@ -0,0 +1,72 @@
+use std::env::args;
+use structopt::clap::AppSettings;
+use structopt::StructOpt;
+
+use crate::commands::{self, Command};
+
+#[derive(StructOpt)]
+#[structopt(
+    name = "Apollo CLI",
+    about = "The [Experimental] Apollo CLI, for supporting all your graphql needs",
+    rename_all = "kebab-case",
+    global_setting = AppSettings::ColoredHelp,
+    global_setting = AppSettings::ColorAuto,
+    global_setting = AppSettings::DeriveDisplayOrder,
+    global_setting = AppSettings::DontCollapseArgsInUsage,
+    global_setting = AppSettings::VersionlessSubcommands,
+)]
+/// The [Experimental] Apollo CLI, for supporting all your graphql needs :)
+pub struct Apollo {
+    #[structopt(subcommand)]
+    pub command: Option<Subcommand>,
```

Comment:
Oo this is an interesting approach! I was gonna suggest we name this `Command` instead of `Subcommand` but that's already taken 🙃

Reply:
We could also rename the trait if we like